### PR TITLE
[GitHub Actions] fix clang compilation

### LINKED ISSUE #4517: [GitHub Actions] Clang compilation does not rely on clang
Reported by Nathan Chancelor, the use of environment variables for CC=clang does not work.

````bash
export CC=clang
make
````
does not work

````bash
make CC=clang
````
works.

But merge_config.sh does not work with either...

### PR DESCRIPTION
**Pull request by plbossart**
Using 'export CC=clang' does not quite work, we have to explicitly use 'make CC=clang'

Link: https://lore.kernel.org/CAK7LNAT6Yp3oemUxSst+htnmM-St8WmSv+UZ2x2XF23cw-kU-Q@mail.gmail.com/
Link: https://lore.kernel.org/alsa-devel/20230809200906.GA4016444@dev-arch.thelio-3990X/T/#t
Closes: https://github.com/thesofproject/linux/issues/4517
Suggested-by: Nathan Chancellor <nathan@kernel.org>

### CODE REVIEW

**Review by bardliao — DISMISSED** [review]

File: sound/soc/sof/intel/hda.c [hunk 1]
```diff
@@ -420,6 +420,7 @@ static bool hda_sdw_check_wakeen_irq(struct snd_sof_dev *sdev)
 	if (!(interface_mask & BIT(SOF_DAI_INTEL_ALH)))
 		return false;
 
+	chip = get_chip_info(sdev->pdata);
```
Comment (ujfalusi) [inline sound/soc/sof/intel/hda.c:423]: :astonished:

**Review by ujfalusi — APPROVED** [review]

File: sound/soc/sof/intel/hda.c (hunk 1 above)
Comment (dbaluta) [reply sound/soc/sof/intel/hda.c:423]: I'm puzzled that this worked until now :)

Comment (ujfalusi) [reply sound/soc/sof/intel/hda.c:423]: yeah, me too
and most importantly how does this slipped through...

Comment (ujfalusi) [reply sound/soc/sof/intel/hda.c:423]: I mean: gcc, sparse should have spotted this with ease, but it is not. Not to mention the reviewers!

Comment (plbossart) [reply sound/soc/sof/intel/hda.c:423]: yeah two big fails for the prize of one
a) the jack detection was completely broken and we didn't notice and
b) we didn't even compile with clang.

I don't know how we are going to send people on Mars ...

**Pull request merged**